homework_1/task3/task3.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/sethgrid/pester"
+)
+
+type pokemonAPIResponse struct {
+	Name                     string
+	Location_area_encounters string
+}
+
+type pokemonEncounters struct {
+	Location_area struct {
+		Name string
+	}
+}
+
+type pokemonStruct struct {
+	Name     string
+	Location []string
+}
+
+func parseProgramArguments(pokemon *string) {
+	flag.StringVar(pokemon, "pokemon", "1", "desired pokemon's name or number")
+
+	flag.Parse()
+}
+
+func fetchHTTPResponse(url string) (*http.Response, error) {
+	httpClient := pester.New()
+	return httpClient.Get(url)
+}
+
+func getResponseContent(response *http.Response) ([]byte, error) {
+	return ioutil.ReadAll(response.Body)
+}
+
+func getDataFromURL(url string) []byte {
+
+	httpResponse, err := fetchHTTPResponse(url)
+	if err != nil {
+		log.Fatal(
+			errors.WithMessage(err, "fetching data from pokemon API"),
+		)
```

Comment:
`log.Fatal` shuts the whole program down (by calling `os.Exit(1)`). This could potentially be dangerous when done from within a non-main function. General suggestion - if a function doesn't know how to handle an error, return it to the caller.

In this example `getDataFromURL` should simply do `return nil, errors.WithMessage(err, "fetching data from pokemon API")` and then `main` should receive this error and handle it (by calling `log.Fatal`).